### PR TITLE
Added some more options: Trafo_variables which transforms ProbNN vari…

### PR DESCRIPTION
**Pull request by burneyy**
…ables in downloaded data, arguments as --config to existing options, optimised PID binnings, and changed 'tree_key' to 'key' to be compatible with recent root_pandas version

### CODE REVIEW

**Comment by KonstantinSchubert** [comment]
Thank you for your pull request. 

Can you please make the pull request to the `development` branch instead of master?

Cheers,
Konstantin

File: pidtool.py [hunk 1]
```diff
@@ -112,43 +136,61 @@ def create_resamplers(options):
...
-            resampler_location = '{particle}_Stripping{stripping}_Magnet{magnet}.pkl'.format(**sample)
+            resampler_location = options.saveto + '/{particle}_Stripping{stripping}_Magnet{magnet}.pkl'.format(**sample)
         if os.path.exists(resampler_location):
             os.remove(resampler_location)
         resamplers = dict()
         deps = map(lambda x: x.format(sample['branch_particle']), kin_variables)
         pids = map(lambda x: x.format(sample['branch_particle']), pid_variables)
         for pid in pids:
             if "DLL" in pid:
-                target_binning = np.linspace(-150, 150, 300) # binning for DLL
```
Comment (KonstantinSchubert) [inline pidtool.py:144]: I don't think these elaborate binning heuristics do not belong into the main resampling module.

Instead we should make it easier for the user to pass a user-defined binning. 

Maybe we should remove the whole RooBinning import/conversion stuff from the module and put it in its own file into a `tools` subdir.

Instead we could have a binning_config.py where the binning can be optionally configured (or imported from RooBinning)

**Comment by KonstantinSchubert** [comment]
Also, I am not convinced that the 

```
def transform_variables(options)
```

method belongs into the package. It doesn't depend on anything else in the module and could easily be kept separate. Maybe it could get its own file and go into a `tools` directory.

**Comment by KonstantinSchubert** [comment]
I think this pull request contains a lot of good ideas but it also brings a lot of stuff that is too specific in my opinion.

I suggest we turn the good ideas into enhancement-issues and discuss / implement them in this way. What do you think @ibab ?

**Comment by ttekampe** [comment]
Fixed by #22

**Pull request closed without merging**